### PR TITLE
feat: add fullReadJson option

### PR DESCRIPTION
**Pull request by fritzy**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -9252,20 +9654,36 @@
           "bundled": true,
           "dev": true
         },
+        "react": {
```
Comment (SchnWalter) [inline package-lock.json]: Is it really neccessary for such an essential library to have a dependency on React? This basically introduces a ton of attac surface.

Comment (nlf) [reply package-lock.json]: it's a dev dependency, it won't be installed for consumers of this package.

File: lib/dir.js [hunk 2]
```diff
@@ -3,12 +3,12 @@ const FileFetcher = require('./file.js')
 const cacache = require('cacache')
 const Minipass = require('minipass')
 const { promisify } = require('util')
-const readPackageJson = require('read-package-json-fast')
 const tarCreateOptions = require('./util/tar-create-options.js')
 const packlist = require('npm-packlist')
 const tar = require('tar')
 const _prepareDir = Symbol('_prepareDir')
 const { resolve } = require('path')
+const _readPackageJson = Symbol.for('package.Fetcher._readPackageJson')
```
Comment (wraithgar) [inline lib/dir.js:11]: This seems odd

Comment (wraithgar) [reply lib/dir.js:11]: Ah looking at the other file that's just how they're formatted

File: lib/fetcher.js [hunk 3]
```diff
@@ -93,6 +97,12 @@ class FetcherBase {
 
     this.before = opts.before
     this.fullMetadata = this.before ? true : !!opts.fullMetadata
+    this.fullReadJson = !!opts.fullReadJson
```
Comment (wraithgar) [inline lib/fetcher.js:100]: Let's at least add this new opts attribute to the readme

Comment (wraithgar) [reply lib/fetcher.js:100]: Ah. It's already there. We're just ... using it now.

This is a bugfix lol

Comment (fritzy) [reply lib/fetcher.js:100]: done

File: package-lock.json (hunk 1 above)
Comment (SchnWalter) [reply package-lock.json:9657]: Yes, it's a dev dependency, but core packages like this one should depend on simpler test frameworks and should not pull down so many packages, just for testing. This makes the ecosystem that more fragile and more prone to incidents like the one that "affected" colors and faker pacages.

Also, if we look at the non-dev depedencies, with this PR, the package now has a dependency on the `has` package, which is described as:

> Object.prototype.hasOwnProperty.call shortcut

And that package actually depends on another basic one: `function-bind`. All of these could have been replaced with a call to `hasOwnProperty` and with a guard agains undefined and null values.

I'm not saying that NPM should reinvent the wheel in all cases, but it least, it should work on streamlining the dependencies added to the core packages.

**Review by wraithgar — APPROVED** [review]

**Pull request merged**